### PR TITLE
Renamed Hashes.Hash256 to Hashes.DoubleSHA256, to make it clear what the method actually does

### PR DESCRIPTION
**Pull request by albracko**
Recently i was trying to verify some payload with the signature and lost a few hours because i wasn't aware that Hash256 method is actually doing a double hash, because the name of the method suggests that it does a single hash.

### CODE REVIEW

**Comment by nopara73** [comment]
Concept ACK.

File: NBitcoin/Crypto/Hashes.cs [hunk 1]
```diff
@@ -18,24 +18,24 @@ namespace NBitcoin.Crypto
 {
 	public static class Hashes
 	{
-		#region Hash256
-		public static uint256 Hash256(byte[] data)
+		#region DoubleSHA256
+		public static uint256 DoubleSHA256(byte[] data)
```
Comment (knocte) [inline NBitcoin/Crypto/Hashes.cs:22]: this is breaking API; you should leave the old methods there but with an `[Obsolete]` attribute

Comment (albracko) [reply NBitcoin/Crypto/Hashes.cs:22]: Agreed...will change it, to be backward compatible

**Comment by NicolasDorier** [comment]
We can't just rename methods, it will break consumers of NBitcoin.

What I advise you is to keep this commit, but add back the `Hash256` method with an `[Obsolete]` argument explaining it is deprecated and use the DoubleSHA256 instead.
I will clean up the old method after one year or two.

**Comment by knocte** [comment]
@NicolasDorier I already told the same to @albracko and he already addresses the issue in his last push ;)

**Comment by NicolasDorier** [comment]
oops indeed sorry!

**Pull request merged**